analysis/src/romgen.rs

```diff
@@ -0,0 +1,62 @@
+use ast::asm_analysis::{AnalysisASMFile, Machine, Program};
+use number::FieldElement;
+use std::marker::PhantomData;
+
+/// Generate the ROM for each machine based on its operations
+
+pub fn generate_rom<T: FieldElement>(file: AnalysisASMFile<T>) -> AnalysisASMFile<T> {
+    RomGenerator::default().generate(file)
+}
+
+#[derive(Default)]
+struct RomGenerator<T> {
+    marker: PhantomData<T>,
+}
+
+impl<T: FieldElement> RomGenerator<T> {
+    fn generate(&self, file: AnalysisASMFile<T>) -> AnalysisASMFile<T> {
+        AnalysisASMFile {
+            machines: file
+                .machines
+                .into_iter()
+                .map(|(name, m)| (name, self.generate_machine_rom(m)))
+                .collect(),
+        }
+    }
+
+    fn generate_machine_rom(&self, mut machine: Machine<T>) -> Machine<T> {
+        machine.program = machine.has_pc().then(|| {
+            assert_eq!(
+                machine.operations.len(),
+                1,
+                "only a single operation is supported"
```

Comment:
Add a comment at the beginning of the file explaining why this romgen is this simple